File: CSharp/Tests/CountOfCharactersTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class CountOfCharactersTest
    {
        [Theory]
        [InlineData("AB", "CD", true)]
        [InlineData("ABC", "DE", false)]
        [InlineData("hello", "edabit", false)]
        [InlineData("meow", "woof", true)]
        [InlineData("jrnvjrnnt", "cvjknfjvmfvnfjn", false)]
        [InlineData("jkvnjrt", "krnf", false)]
        [InlineData("Facebook", "Snapchat", true)]
        public void Compare_TwoStringValues_ReturnBoolean(string str1, string str2, bool expected)
        {
            var actual = CountOfCharacters.Compare(str1, str2);

            Assert.Equal(expected, actual);
        }
    }
}

File: js/tests/countOfCharacters.test.js
const compare = require("../countOfCharacters");

describe("compare", () => {
  describe("Return true if two string have the same length", () => {
    const data = [
      ["AB", "CD"],
      ["meow", "woof"],
      ["Facebook", "Snapchat"]
    ];

    test.each(data)("compare(%s, %s) = true", (str1, str2) => {
      var actual = compare(str1, str2);

      expect(actual).toEqual(true);
    });
  });

  describe("Return false if two strings do not have the same length", () => {
    const data = [
      ["ABC", "DE"],
      ["hello", "edabit"],
      ["jrnvjrnnt", "cvjknfjvmfvnfjn"],
      ["jkvnjrt", "krnf"]
    ];

    test.each(data)("compare(%s, %s) = false", (str1, str2) => {
      var actual = compare(str1, str2);

      expect(actual).toEqual(false);
    });
  });
});
